src/bors/merge_queue.rs

```diff
@@ -34,6 +36,10 @@ enum MergeResult {
 }
 
 pub async fn handle_merge_queue(ctx: Arc<BorsContext>) -> anyhow::Result<()> {
+    // Prevent concurrent merge queue processing.
+    let lock = MERGE_QUEUE_LOCK.get_or_init(|| Mutex::new(()));
```

Comment:
Do we actually need the lock for anything right now? It seems like if we made this function private and exposed just a future that runs a loop that reads from a channel, it should be impossible to invoke the merge queue concurrently, which would be much nicer than using an actual lock.